src/eigen/diag.jl

```diff
@@ -50,7 +50,7 @@ function diagonalise_all_kblocks(eigensolver, ham::Hamiltonian, nev_per_kpoint::
         @assert size(guessk) == (length(kpoints[ik].basis), nev_per_kpoint)
 
         results[ik] = eigensolver(kblock(ham, kpt), guessk;
-                                  prec=kblock(prec, kpt), tol=tol,
+                                  prec=prec(ham, kpt), tol=tol,
```

Comment:
Puh ... that took me a second to understand. Could you make a comment to make sure that before `prec` is a type and after an instance.